iac-extensions/kubernetes/src/main/java/org/sonar/iac/helm/tree/utils/GoTemplateAstHelper.java

```diff
@@ -38,25 +41,43 @@ private GoTemplateAstHelper() {
     // utility class
   }
 
-  public static Stream<FieldNode> findValuePathNodes(GoTemplateTree tree, TextRange range, String sourceText) {
+  public static Stream<Node> findNodesToHighlight(GoTemplateTree tree, TextRange range, String sourceText) {
     var location = LocationImpl.fromTextRange(range, sourceText);
+
+    return Stream.concat(
+      findValuePathNodes(tree, location),
+      findIncludeFunctionsFirstArg(tree, location));
+  }
+
+  static Stream<FieldNode> findValuePathNodes(GoTemplateTree tree, Location location) {
     var nodes = tree.root().children().stream()
       .filter(hasOverlayingLocation(location))
       .toList();
 
-    return allChildren(nodes).stream()
+    return nodesWithImmediateChildren(nodes).stream()
       .filter(FieldNode.class::isInstance)
+      // Sometimes top-level nodes have a very broad range and some children can be actually outside the range.
       .filter(hasOverlayingLocation(location))
       .map(FieldNode.class::cast);
   }
 
   public static List<ValuePath> findValuePaths(GoTemplateTree tree, TextRange range, String sourceText) {
-    return findValuePathNodes(tree, range, sourceText)
+    var location = LocationImpl.fromTextRange(range, sourceText);
+
+    return findValuePathNodes(tree, location)
       .map(FieldNode::identifiers)
       .map(ValuePath::new)
       .toList();
   }
 
+  private static Stream<Node> findIncludeFunctionsFirstArg(GoTemplateTree tree, Location location) {
+    return collectNodesByType(tree.root(), NodeType.NODE_COMMAND, CommandNode.class)
+      .filter(hasOverlayingLocation(location))
+      .filter(cmd -> !cmd.arguments().isEmpty())
```

Comment:
This is not a good filter when accessing more than one element of the list. `cmd.arguments().size() > 1` should be more safe, even when it is slower. Currently, you assume, that when there is at least one element, there should be the 2nd as well. This may be the case right now, but maybe not in the future.